consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftBlockHeightManager.java

```diff
@@ -229,24 +256,30 @@ public void handleRoundChangePayload(final RoundChange message) {
       final RoundChangeArtifacts roundChangeMetadata = RoundChangeArtifacts.create(result.get());
 
       if (finalState.isLocalNodeProposerForRound(targetRound)) {
-        currentRound.startRoundWith(
-            roundChangeMetadata, TimeUnit.MILLISECONDS.toSeconds(clock.millis()));
+        if (currentRound.isEmpty()) {
```

Comment:
I don't think that this `if` block should be here.

Reply:
It is ok to move from "no round" to any round as a consequence of a quorum of RoundChange messages

Reply:
This doesn't prevent moving from no round to a round. The if block is just so that we have a QbftRound object to call startRoundWith if this happens before the block expiry has occurred.